docs/launcher.md

```diff
@@ -0,0 +1,107 @@
+
+`adb shell "am start -n com.airbnb.sample/com.airbnb.mvrx.launcher.MavericksLauncherActivity --es viewNameToOpen 'Default state'"`
+
+You may also find it helpful to define functions in your bash profile to make it easier to call these.
+
+```
+# Open the launcher
+function mvrx_launcher() {
+  adb shell am start -n com.your.application.package/com.airbnb.mvrx.launcher.MvRxLauncherActivity
+}
+
+# Test fragments and mocks that match the given pattern.
+# ie: mvrx_test my_fragment
+function mvrx_test() {
+  adb shell am start -n com.airbnb.sample/com.airbnb.mvrx.launcher.MvRxLauncherActivity --es viewNamePatternToTest $1
+}
+
+# Test all mocks in the app
+function mvrx_test_all() {
+  adb shell am start -n com.airbnb.sample/com.airbnb.mvrx.launcher.MvRxLauncherActivity --es viewNamePatternToTest ""
+}
+
+# Open the first fragment that matches the pattern
+# ie: mvrx_open MyFragment
+function mvrx_open() {
+  adb shell am start -n com.airbnb.sample/com.airbnb.mvrx.launcher.MvRxLauncherActivity --es viewNameToOpen $1
+}
+```
+
+### Jetpack Navigation Support
+
+By default the launcher does not work in an app configured for jetpack navigation, since it cannot know how to start a fragment with your navigation graph. You can define a custom activity with your nav graph that the launcher will use instead. See LauncherActivity in the mvrx sample module for an example.
+
+## Automated testing of mocks from the launcher
+
+From the Launcher activity a "Test All" button is available in the Fragment toolbar that, when clicked, will open each mock in series to check for crashes upon initialization.
+
+This offers a sanity check that each mock can be opened and loaded without crashes.
+
+By default this opens each mock in the `MavericksLauncherTestMocksActivity` activity, but that behavior can be changed via setting a custom intent for `MavericksLauncherTestMocksActivity.provideIntentToTestMock`
+
+The testing offered by this is fairly naive and basic, but could be expanded with future work.
+
+## Caveats
+Since each mocked fragment is launched in isolation it must be able to function independently, without assuming any other fragments or activities are present. For example, it must not make assumptions about what its parent fragment or parent activity is.
+
+This means that if you need inter-fragment communication you cannot assume another fragment exists, but instead must use decoupled approaches, such as a Dagger injected class.
+
+Note that the same restriction applies for fragments being run in automatic test suites, so all Mavericks views should follow this.
+
+## How It Works
+Mavericks inspects the app Dex files to find all mockable Mavericks views and their declared mock states and arguments. This process can take some time, so Fragment and mock information is saved in shared preferences for faster loading on subsequent app starts.
```

Comment:
Is this worth an investigation to datastore down the road? If so maybe remove the specific technology we use the store the data